### PR TITLE
chore: add permissive LICENSE.md

### LINKED ISSUE #116: Missing LICENSE
We need to 

- [x] Port dual-licensing we do in Kubo:
  - https://github.com/ipfs/kubo/blob/master/README.md?plain=1#L433-L438
  - https://github.com/ipfs/kubo/blob/master/LICENSE
- [x] Get sign-off from everyone that did commits so far

### PR DESCRIPTION
**Pull request by 2color**
Fixes #116

### CODE REVIEW

**Comment by SgtPooki** [comment]
We should use the same pattern as libp2p with two license files

File: LICENSE.md
Comment (lidel) [inline LICENSE.md:1]: @2color any reason why we used different syntax than we have in Kubo and Helia repos?

I was under impression we want to copy&paste:
- https://github.com/ipfs/helia/blob/main/LICENSE (we could add SPDX tags here and in README)
- https://github.com/ipfs/helia/blob/main/LICENSE-MIT
- https://github.com/ipfs/helia/blob/main/LICENSE-APACHE

**Review by lidel — CHANGES_REQUESTED** [review]
@2color thanks! Just a small concern around making this repo use different "syntax" than Kubo and Helia. 

Is the idea to update them? If not, see comment below.

File: LICENSE.md
Comment (aschmahmann) [reply LICENSE.md:1]: We can use a single one, no https://github.com/ipfs/ipfs-repository-template/blob/c0e7956eae1c7541e11fc24c24058085fc8f3339/LICENSE.md? If not should we change the repo template?

Comment (SgtPooki) [reply LICENSE.md:1]: js-libp2p and helia have been using separate ones for a while, we should stick to that pattern.

Comment (2color) [reply LICENSE.md:1]: I don't have strong opinions. I also don't want to waste time on this.

I've stuck to the same convention as Helia and added the SPDX tag in the `LICENSE` file. 

Hopefully that should keep everyone happy.

**Review by SgtPooki — APPROVED** [review]

**Comment by SgtPooki** [comment]
@lidel, what should we do for the dev sign-off process for previous committers?

**Comment by 2color** [comment]
I think we can just continue collecting approvals here. 

@lidel @whizzzkid @lidel @achingbrain @MarcoPolo @aschmahmann 

Can you please review this and approve to confirm you are ok with the license update?

**Review by lidel — COMMENTED** [review]
I accept these license terms.

ps. we should get approvals before merging, but as long everyone replies here we are good and don't have to revert :)

**Comment by SgtPooki** [comment]
I accept these license terms.

**Comment by aschmahmann** [comment]
I accept these license terms.

**Comment by whizzzkid** [comment]
I accept these license terms.

**Comment by achingbrain** [comment]
I accept these license terms.

File: LICENSE.md
Comment (achingbrain) [reply LICENSE.md:1]: Urgh, what a pain.  aegir-managed projects should be using these files: https://github.com/ipfs/aegir/tree/master/src/check-project/files which I think were taken from go-ipfs in the dim and distant past.  I may be wrong, memory is a funny thing.

**Comment by SgtPooki** [comment]
just need Marco and I think we're good

**Comment by MarcoPolo** [comment]
I accept these license terms.

File: LICENSE.md
Comment (2color) [reply LICENSE.md:1]: @achingbrain what implication does it have if the license files are different?

For example, in this repo I added the `SPDX-License-Identifier: Apache-2.0 OR MIT` tag to `LICENSE`.

**Pull request merged**
